core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -655,14 +656,21 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
     *
     * @return A brokerId.
     */
-  private def getBrokerId: Int =  {
+  private def getBrokerIdAndOfflineDirs: (Int, Seq[String]) =  {
```

Comment:
Ah I should have done this. Thanks for noticing this. I have updated the patch with the following comment:

```
    * The log directories whose meta.properties can not be accessed due to IOException will be returned to the caller
    * @return A 2-tuple containing the brokerId and a sequence of log directories.
```